adapter/sql/util_test.go

```diff
@@ -28,3 +28,57 @@ func TestToInt64(t *testing.T) {
 	assert.Equal(t, int64(1), toInt64(uint16(1)))
 	assert.Equal(t, int64(1), toInt64(uint8(1)))
 }
+
+func TestEscape(t *testing.T) {
+	config := Config{
+		Placeholder: "?",
+		EscapeChar:  "`",
+	}
+
+	tests := []struct {
+		field  string
+		result string
+	}{
+		{
+			field:  "count(*) as count",
+			result: "count(*) AS `count`",
+		},
+		{
+			field:  "count(*) as `count`",
+			result: "count(*) AS `count`",
+		},
+		{
+			field:  "count(*) as count", // duplicated test to simulate cache load
+			result: "count(*) AS `count`",
+		},
+		{
+			field:  "user.address as home_address",
+			result: "`user`.`address` AS `home_address`",
+		},
+		{
+			field:  "^FIELD(`gender`, \"male\") AS order",
```

Comment:
This behavior shouldn't be changed, when using `^` as the first character, the field should not be escaped